pkg/cmd/task/delete.go

```diff
@@ -81,6 +81,12 @@ func deleteTask(opts *deleteOptions, s *cli.Stream, p cli.Params, tName string)
 		return fmt.Errorf("failed to create tekton client")
 	}
 
+	// Check if namespace exists. Return error if namespace specified with -n doesn't exist or if user doesn't have permissions to view.
```

Comment:
imho, these validations should be done before calling `checkOptions` (asks for user input) otherwise the UX will be that they issue a `task delete -n foobar task-name` and `checkOptions` would ask for confirmation and then fail saying there is no namespace `foobar` where as we we could do the validation before invoking `checkOptions` it would fail soon after they run the `task delete -n foobar ....` .
 
I am "fine" with this implementation but if you feel motivated, could you consider moving the validations to a different function?

Reply:
@sthaha Yeah sure. I see your point and think it makes sense. So basically move the validation up to before `checkOptions`? 

I don't think that should change the behavior here, and I can also correct this then for [`list.go`](https://github.com/tektoncd/cli/blob/aea08692f3b6d4fdc81866d456157ea17247e491/pkg/cmd/task/list.go#L80) since we already merged in this validation for `tkn task list`.

Reply:
I pushed the validation before `checkOptions` in latest push, but feel free to still recommend a better location.